src/main/java/com/cloudbees/jenkins/support/impl/SlaveCommandStatistics.java

```diff
@@ -56,7 +65,14 @@
 @Extension
 public final class SlaveCommandStatistics extends Component {
 
+    /*protected*/ static @Nonnegative int MAX_STATS_SIZE = 1000;
+
+    private final Object statLock = new Object();
```

Comment:
Simpler to just synchronize on `this` (e.g., `synchronized` method modified).

Reply:
I prefer to have the lock encapsulated so no unrelated code would share the same monitor accidentally - which happens to the intrinsic one. Do you feel strongly against?

Reply:
Not at all, just a thought. If I “felt strongly” I would request changes!